### PR TITLE
Add Recent Messages privacy disclaimer.

### LINKED ISSUE #1311: Create opt-in system for recent-messages and other third-party services
**Is your feature request related to a problem? Please describe.**
I'm trying to make my recent_messages service more compliant with the law

**Describe the solution you'd like**
A section in the installer where the user is shown
- that Chatterino2 can fetch messages from the recent_messages service
- it will send each channel name you visit to a third-party service for that,
- and that channels you visit will have their messages stored
- that you can opt out of having the bot collect messages in your own channel by using the bot at https://www.twitch.tv/recent_messages
and with a link to the privacy statement (`https://recent-messages.robotty.de/privacy`), and two buttons at the bottom "Accept terms and use service", "Decline terms and don't use service" (or something like that)

**Describe alternatives you've considered**
Don't do it i guess, but it would be nice, also to keep the users more informed

**Additional context**

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description
Adds a disclaimer about the recent messages service. Closes #1311.

### CODE REVIEW

File: src/Application.cpp [hunk 1]
```diff
@@ -137,6 +137,56 @@ int Application::run(QApplication &qtApp)
...
+        auto reply = QMessageBox::question(
+            &this->windows->getMainWindow(),
+            "Do you want to enable Recent Messages support?",
+            "Recent messages is a service which allows you to see "
+            "messages from before starting Chatterino. If you enable support "
+            "for it Chatterino will send the name of every channel you have "
+            "open for the purposes of showing you the message history. You can "
+            "learn more <a "
+            "href=\"https://recent-messages.robotty.de/\">here</a> or "
+            "<a href=\"https://recent-messages.robotty.de/privacy\">view the "
```
Comment (23rd) [inline src/Application.cpp:151]: What if the `CHATTERINO2_RECENT_MESSAGES_URL` environment variable will be replaced?

File: src/Application.cpp [hunk 2]
```diff
@@ -137,6 +137,56 @@ int Application::run(QApplication &qtApp)
...
+                        }
+                    }
+                }
+            }
+            getSettings()->loadTwitchMessageHistoryOnConnect = true;
+        }
+        else
+        {
+            getSettings()->loadTwitchMessageHistoryOnConnect = false;
+        }
```
Comment (23rd) [inline src/Application.cpp:186]:
```suggestion
        const auto isConfirmed = (reply == QMessageBox::Yes);
        if (isConfirmed)
        {
            if (!getSettings()->loadTwitchMessageHistoryOnConnect)
            {
                if (auto selected = this->windows->getMainWindow()
                                        .getNotebook()
                                        .getSelectedPage())
                {
                    if (auto container =
                            dynamic_cast<SplitContainer *>(selected))
                    {
                        for (auto &&split : container->getSplits())
                        {
                            if (auto channel = split->getChannel();
                                !channel->isEmpty())
                            {
                                channel->addMessage(makeSystemMessage(
                                    "You will need to restart chatterino to "
                                    "see historical messages."));
                            }
                        }
                    }
                }
            }
        }
        getSettings()->loadTwitchMessageHistoryOnConnect = isConfirmed;
```

File: src/Application.cpp [hunk 3]
```diff
@@ -137,6 +137,56 @@ int Application::run(QApplication &qtApp)
...
     this->windows->getMainWindow().show();
+    if (!getSettings()->seenRecentMessagesDisclaimer)
+    {
+        auto reply = QMessageBox::question(
+            &this->windows->getMainWindow(),
+            "Do you want to enable Recent Messages support?",
+            "Recent messages is a service which allows you to see "
+            "messages from before starting Chatterino. If you enable support "
+            "for it Chatterino will send the name of every channel you have "
+            "open for the purposes of showing you the message history. You can "
```
Comment (RAnders00) [inline src/Application.cpp:148]:
```suggestion
            "open to a remote service for the purposes of showing you the message history. You can "
```

File: src/Application.cpp (hunk 1 above)
Comment (Mm2PL) [reply src/Application.cpp]: Changed in 78ab484, links are now dynamic based on the URL given from the ENV variable.

**Comment by leon-richardt** [comment]
Should we also mention in the info text that the external service can see your IP? (In distinction to Chatterino API requests where we act as a proxy so external providers cannot see user IPs.)

**Comment by pajlada** [comment]
This sort of messaging should probably be part of the installers instead.

However, a privacy notice of some kind could be part of the About page describing the third party APIs we consume. This PR could be the first third party API to be added to the About page

**Comment by Mm2PL** [comment]
Yes it should be a part of the installer, however the portable versions don't have an installer, so it would show up only for some users, not a good idea. The idea of this popup was to ask for consent for sending channel data.

**Comment by Mm2PL** [comment]
This is getting stale. I might reimplement this some time in the future but it isn't now.

**Pull request closed without merging**